optimism/src/mips/column.rs

```diff
@@ -74,3 +66,27 @@ impl<T: Clone> MIPSWitnessTrait<T> for MIPSWitness<T> {
         &self.cols[SCRATCH_SIZE + 1]
     }
 }
+
+impl<T: Clone> Index<Column> for MIPSWitness<T> {
+    type Output = T;
+
+    /// Map the column alias to the actual column index.
+    /// Note that the column index depends on the step kind (Sponge or Round).
+    /// For instance, the column 800 represents PadLength in the Sponge step, while it
+    /// is used by intermediary values when executing the Round step.
+    fn index(&self, index: Column) -> &Self::Output {
+        match index {
+            Column::ScratchState(i) => &self.scratch()[i],
```

Comment:
it is different in the implementation below, just highlighting, it is likely fine.